contribute/breaking-changes-guide.md

```diff
@@ -0,0 +1,120 @@
+possible breaking changes. It also adds the `"breaking change"` label to the pull request.
+
+## How does the CI workflow look like?
+
+<img src="./breaking-changes-workflow.png" alt="CI workflow" width="700" />
+
+## I received a comment on my PR, what does it mean?
+
+![Levitate comment](./breaking-changes-comment-screenshot.png)
+
+Receiving a comment like the one above does not necessarily mean that you actually introduced breaking
+changes (as certain edge cases are still not covered by the tool), but as there is a good chance we rather raise attention.
+
+By clicking the links in the comment ("more info" or "Check console output") you can see the actual console
+output of the tool which gives more detailed information on what triggered the notification.
+
+**Removed exported members** (consolve view):<br />
+This means that some previously exported members won't be available in the newer version of the package which can break dependent plugins.
+
+![](./breaking-changes-console-screenshot-1.png)
+
+**Changed an existing member** (consolve view):<br />
+This means that a member was changed in a way that can break dependent plugins.
+
+![](./breaking-changes-console-screenshot-2.png)
+
+**No breaking changes** (consolve view):<br />
+Seeing this does not mean that you made no changes, but that most probably none of them were breaking. You are good to go! 👏
+
+![](./breaking-changes-console-screenshot-3.png)
+
+## How can I decide if it is really a breaking change?
+
+First go to the console output of the workflow and make sure that the diffs make sense.
+
+It can happen that Levitate highlights a change which is marked with TSDoc tags `// @alpha` or `// @internal` in
+which case you can choose to ignore it - keep in mind though that these flags won't really hold developers back
+from using your code and most likely it is going to cause them problems if we are breaking them.
+
+It can also happen that Levitate marks changing an interface as a possible breaking change.
+While it is correct (if anyone implements that interface introducing a new property will break their code),
+maybe it is an interface that is basically never implemented by other developers, in which case you
+can choose to ignore Levitate's message.
+
+These notifications are only warnings though, and **in the end it's up to the author of the PR to make a decision that makes the most sense.**
+
+## I know it's a breaking change, what's next?
+
+### Introduce breaking changes only in major versions
+
+We can make breaking changes less painful if they are only happening between major releases of Grafana.
+
+### Deprecate first
+
+Whenever possible try to deprecate first what you are about to remove or change. For example:
+
+```javascript
+/**
+ * @deprecated -- this is no longer necessary and will be removed in Grafana 9.0.0
+ */
+myOldFunction(name: string) {
+    deprecationWarning('MyFile', 'myOldFunction', 'myNewFunction');
+    // ...
+}
+```
```

Comment:
```suggestion
```javascript
import { deprecationWarning } from '@grafana/data';

/**
 * @deprecated -- this is no longer necessary and will be removed in Grafana 9.0.0
 */
myOldFunction(name: string) {
    deprecationWarning('MyFile', 'myOldFunction', 'myNewFunction');
    // ...
}
```
```

Reply:
Amazing, thanks! 🙏